packages/cli/src/platform/login.ts

```diff
@@ -1,12 +1,111 @@
-import { Command } from '@prisma/internals'
+import { Command, isError } from '@prisma/internals'
+import listen from 'async-listen'
+import http from 'http'
+import { underline } from 'kleur/colors'
+import open from 'open'
+
+import { getInstalledPrismaClientVersion } from '../utils/getClientVersion'
+import { platformConsoleUrl, writeAuthConfig } from '../utils/platform'
+
+interface AuthResult {
+  token: string
+  user: {
+    id: string
+    displayName: string
+    handle: string
+    email: string
+  }
+}
 
 export class Login implements Command {
   public static new(): Login {
     return new Login()
   }
 
-  public async parse(argv: string[]) {
-    await Promise.resolve('todo')
-    return JSON.stringify(argv)
+  public async parse() {
+    console.log('Authenticating to Prisma Platform CLI via browser')
+
+    const server = http.createServer()
+    const { port } = await listen(server, 0, '127.0.0.1')
```

Comment:
I'm a bit confused by the parameters here for `listen`, the documentation of the library is quite vague, and this options are not included there, would you mind adding some docs explaining what the `0` means for example?

On a side note, would it be better to just use the result instead for the full url?
I'm getting this snippet from the [library docs](https://www.npmjs.com/package/async-listen):

```ts
const address = await listen(server, port);
console.log(address); // => URL('http://127.0.0.1:3000')

// use address instead of concatenate the port later 
```

Reply:
Will update!